Improve CLI

jan-machek-ds opened the pull request:
Commands now write out valid help and can also be called as space-separated subcommands in addition to the default semicolon-separated style.

jan-machek-ds committed:
Improve CLI

Commands now write out valid help and can also be called as space-separated subcommands in addition to the default semicolon-separated style.

kutny commented on src/consolebundle/CommandRunner.py:
```diff
@@ -20,13 +26,15 @@ def run_command():
-            logger.info(existing_command.get_command() + " - " + existing_command.get_description())
+        command_manager.log_subcommands(known_args.command_name, logger)
 
         sys.exit(1)
```
Why exist(1) ? It is not an error, isn't it?

jan-machek-ds replied:
Yup, sorry, this was a copy paste.

kutny commented on src/consolebundle/CommandManager.py:
```diff
@@ -9,9 +9,40 @@ def __init__(self, commands: List[ConsoleCommand]):
+                logger.info(
+                    f"{' '.join(existing_command.get_command().split(':'))} - " + existing_command.get_description())
+
+    def get_by_name(self, name: list) -> ConsoleCommand:
```
Why list?

jan-machek-ds replied:
Name is now a list. It is because of the way argparse works. It does not allow optional positional arguments and command_name is a positional argument. 
So if we want to allow both dbx:deploy and dbx deploy commands to be valid, we have to do it like this. So command_name is either ['dbx:deploy'] or ['dbx', 'deploy'] in this case.

kutny replied:
Ok, so let's rename it to name_parts or something like that. name: list is confusing to me

jan-machek-ds replied:
Already done.

kutny commented on src/consolebundle/CommandManager.py:
```diff
@@ -9,9 +9,40 @@ def __init__(self, commands: List[ConsoleCommand]):
 
         raise Exception('No command with name "{}" found'.format(name))
+
+    def command_prefix_only(self, name: list) -> bool:
```
name -> name_parts?

jan-machek-ds replied:
OK.

kutny commented on src/consolebundle/CommandManager.py:
```diff
@@ -9,9 +9,40 @@ def __init__(self, commands: List[ConsoleCommand]):
         return self.__commands
 
-    def get_by_name(self, name: str) -> ConsoleCommand:
+    def log_commands(self, logger):
```
I would propose to move log_commands() and log_subcommands() out of CommandManager...probably to CommandRunner

jan-machek-ds replied:
Ok, moved.

jan-machek-ds committed:
Fix CLI bugs

Logging moved to CommandRunner.
Top-most level help corrected.
Exit codes fixed.
Command names definitions in command specific py files can now be both semicolon- and space-delimited.

kutny commented on src/consolebundle/CommandManager.py:
```diff
@@ -1,17 +1,48 @@
+    POSSIBLE_SEPARATORS = [':', ' ']
+
     def __init__(self, commands: List[ConsoleCommand]):
         self.__commands = commands
```
Creating i map of (norrmalized) command names and commands here in __init__ would simplify the whole code a lot...

jan-machek-ds replied:
Hmm. I cannot seem to see the simplification. Could you please elaborate on this?
